demo-app/src/components/DataAndFriends/PaginationControls/PaginationControls.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Box, Pagination } from 'grommet';
+import { PaginationSummary } from './PaginationSummary';
+import { StepSelector } from './StepSelector';
+
+export const PaginationControls = ({
+  numberItems,
+  page,
+  setPage,
+  step,
+  setStep,
+}) => {
+  return (
+    <Box
+      direction="row"
+      align="center"
+      justify="between"
+      border="top"
+      fill
+      gap="small"
+      pad={{ vertical: 'xsmall' }}
+      wrap
+    >
+      <PaginationSummary numberItems={numberItems} page={page} step={step} />
+      <Box flex="grow" />
+      <StepSelector step={step} setStep={setStep} />
+      <Pagination
```

Comment:
It might be worth trying an approach like demod here where Data is controlling Pagination as opposed to Pagination controlling itself: https://storybook.grommet.io/?path=/story/data-data-spacex--space-x

Not sure how that plays with StepSelector though.

I'm fine leaving as is for now and revisiting in future.